Source/Extensions/Blazorise.DataGrid/_DataGridCellNumericEdit.razor

```diff
@@ -0,0 +1,59 @@
+@using System.Reflection;
+@typeparam TItem
+@inherits ComponentBase
+@if ( valueType == typeof( decimal ) || valueType == typeof( decimal? )
+|| valueType == typeof( double ) || valueType == typeof( double? )
+|| valueType == typeof( float ) || valueType == typeof( float? )
+|| valueType == typeof( int ) || valueType == typeof( int? )
+|| valueType == typeof( long ) || valueType == typeof( long? ) )
+{
+    @numericPickerFragment
+}
+
+
+@code {
+    private RenderFragment numericPickerFragment => builder =>
+       {
+           var type = typeof( NumericPicker<> ).MakeGenericType( new[] { valueType } );
+
+           builder.OpenComponent( 0, type );
+           builder.AddAttribute( 1, nameof( NumericPicker<object>.Value ), valueType switch
+           {
+               Type typeDecimal when typeDecimal == typeof( Decimal ) => (Decimal)( CellValue ?? (Decimal)default ),
+               Type typeDecimalNull when typeDecimalNull == typeof( Decimal? ) => (Decimal?)( CellValue ?? (Decimal?)default ),
+               Type typeDouble when typeDouble == typeof( Double ) => (Double)( CellValue ?? (Double)default ),
+               Type typeDoubleNull when typeDoubleNull == typeof( Double? ) => (Double?)( CellValue ?? (Double?)default ),
+               Type typeFloat when typeFloat == typeof( Single ) => (Single)( CellValue ?? (Single)default ),
+               Type typeFloatNull when typeFloatNull == typeof( Single? ) => (Single?)( CellValue ?? (Single?)default ),
+               Type typeInt when typeInt == typeof( Int32 ) => (Int32)( CellValue ?? (Int32)default ),
+               Type typeIntNull when typeIntNull == typeof( Int32? ) => (Int32?)( CellValue ?? (Int32?)default ),
+               Type typeLong when typeLong == typeof( Int64 ) => (Int64)( CellValue ?? (Int64)default ),
+               Type typeLongNull when typeLongNull == typeof( Int64? ) => (Int64?)( CellValue ?? (Int64?)default ),
+               _ => throw new InvalidOperationException( $"Unsupported type {valueType}" )
```

Comment:
Do we want to throw an exception? It's not how we usually do it.

Reply:
Kinda copied from existing implementations. I mean. It's one of those things, were it's not really expected to hit the exception. If it does, something is very wrong, and it's probably some change in development so it might be better to throw. 
![image](https://github.com/Megabit/Blazorise/assets/22283161/745bc618-ecca-44ac-a7be-37e1463a8f2d)
![image](https://github.com/Megabit/Blazorise/assets/22283161/6ddc6bd7-342a-4340-b6a3-689b19807d1a)

What do you think?

Reply:
I guess it can stay.